lib/rsvp/-internal.js

```diff
@@ -179,27 +202,12 @@ export function publish(promise) {
   promise._subscribers.length = 0;
 }
 
-function ErrorObject() {
-  this.error = null;
-}
-
-const TRY_CATCH_ERROR = new ErrorObject();
-
-function tryCatch(callback, detail) {
-  try {
-    return callback(detail);
-  } catch(e) {
-    TRY_CATCH_ERROR.error = e;
-    return TRY_CATCH_ERROR;
-  }
-}
-
 export function invokeCallback(state, promise, callback, detail) {
   let hasCallback = isFunction(callback);
   let value, error;
 
   if (hasCallback) {
-    value = tryCatch(callback, detail);
```

Comment:
motivation was to make `tryCatch` as following 
* to make it accept list of arguments (instead of one)
* didn't want to slice and dice argument list to extract target/context of callback
* didn't want to create extra functions 
and same approach is used in bluebird.js https://github.com/petkaantonov/bluebird/blob/master/src/util.js#L16